doc/src/manual/control-flow.md

```diff
@@ -842,52 +842,55 @@ them. The consumer cannot simply call a producer function to get a value, becaus
 may have more values to generate and so might not yet be ready to return. With tasks, the producer
 and consumer can both run as long as they need to, passing values back and forth as necessary.
 
-Julia provides the functions [`produce()`](@ref) and [`consume()`](@ref) for solving this problem.
-A producer is a function that calls [`produce()`](@ref) on each value it needs to produce:
+Julia provides a [`Channel`](@ref) mechanism for solving this problem.
+A [`Channel`](@ref) is a waitable FIFO queue which can have multiple tasks reading and writing to it.
+
+Let's define a producer task, which produces values via the [`put!`](@ref) call.
 
 ```julia
-julia> function producer()
-         produce("start")
+julia> function producer(c::Channel)
+         put!(c, "start")
          for n=1:4
-           produce(2n)
+           put!(c, 2n)
          end
-         produce("stop")
+         put!(c, "stop")
        end;
 ```
 
-To consume values, first the producer is wrapped in a [`Task`](@ref), then [`consume()`](@ref)
-is called repeatedly on that object:
+To consume values, we need to schedule the producer to run in a new task. A special [`Channel`](@ref)
+constructor which accepts a 1-arg function as an argument can be used to run a task bound to a channel.
+We can then [`take!()`](@ref) values repeatedly from the channel object:
 
-```julia
-julia> p = Task(producer);
+```jldoctest
```

Comment:
this won't pass unless the previous block has run in the same julia instance to define `producer`